spec/unit/provider/service/base_spec.rb

```diff
@@ -91,4 +91,24 @@ def execute_command(command, options)
       expect {subject.stop }.to raise_error(Puppet::Error, 'Could not stop Service[test]: failed to stop')
     end
   end
+
+  context "when hasstatus is false" do
+    subject do
+      type.new(
+         :name  => "status test",
+         :provider => :base,
+         :hasstatus => false,
+         :pattern => "majestik m\u00f8\u00f8se",
+      ).provider
+    end
+
+    it "retrieves a PID from the process table" do
+      Facter.stubs(:value).with(:operatingsystem).returns("CentOS")
+      ps_output = File.binread(my_fixture("ps_ef.mixed_encoding")).force_encoding(Encoding::UTF_8)
```

Comment:
The output will end up getting mangled. The text fixture was generated on a machine that had a locale set to UTF_8, so this is re-producing that setup in cases when the tests are run under a different locale.

Given the output of `ps` can be in a mixed encoding, my real goal with this PR is to keep regexes containing ASCII characters working instead of throwing errors --- as I suspect that's the majority use case for folks that are using this provider.